frontend/templates/product/sections/CrossSellSection/index.tsx

```diff
@@ -0,0 +1,544 @@
+      setSelectedVariantIds((current) => {
+         if (current.includes(variantId)) {
+            return current.filter((id) => id !== variantId);
+         }
+         return current.concat(variantId);
+      });
+   };
+
+   const totalPrice = React.useMemo(() => {
+      return selectedVariantIds.reduce((acc, id) => {
+         if (id === selectedVariant.id) {
+            return acc + parseFloat(selectedVariant.price.amount);
+         }
+         const variant = selectedVariant.crossSellVariants.find(
+            (variant) => variant.id === id
+         );
+         if (variant) {
+            return acc + parseFloat(variant.price.amount);
+         }
+         return acc;
+      }, 0);
+   }, [
+      selectedVariant.crossSellVariants,
+      selectedVariant.id,
+      selectedVariant.price.amount,
+      selectedVariantIds,
+   ]);
+
+   const formattedTotalPrice = React.useMemo(() => {
+      return formatShopifyPrice({
+         amount: totalPrice,
+         currencyCode: selectedVariant.price.currencyCode,
+      });
+   }, [selectedVariant.price.currencyCode, totalPrice]);
+
+   const currentProduct = React.useMemo<CrossSellProduct>(() => {
+      return {
+         title: product.title,
+         rating: product.reviewsData?.average,
+         reviewsCount: product.reviewsData?.count,
+      };
+   }, [
+      product.reviewsData?.average,
+      product.reviewsData?.count,
+      product.title,
+   ]);
+
+   const handleAddToCart = () => {
+      const input = selectedVariantIds.map(
+         (variantId): AddProductVariantInput | null => {
+            if (variantId === selectedVariant.id) {
+               if (!isPresent(selectedVariant.sku)) {
+                  return null;
+               }
+               return {
+                  name: product.title,
+                  itemcode: selectedVariant.sku,
+                  formattedPrice: selectedVariant.formattedPrice,
+                  quantity: 1,
+                  imageSrc: selectedVariant.image?.url ?? '',
+               };
+            }
+            const variant = selectedVariant.crossSellVariants.find(
+               (v) => v.id === variantId
+            );
+            const variantSku = variant?.sku;
+            if (variant == null || !isPresent(variantSku)) {
+               return null;
+            }
+            return {
+               name: variant.product.title,
+               itemcode: variantSku,
+               formattedPrice: variant.formattedPrice,
+               quantity: 1,
+               imageSrc: variant.image?.url ?? '',
+            };
+         }
+      );
+      const selectedVariantSku = selectedVariant.sku;
+      if (isPresent(selectedVariantSku)) {
+         addToCart.mutate({
+            currentItemCode: selectedVariantSku,
+            items: filterNullableItems(input),
+         });
+         onOpen();
+      } else {
+         console.error('No SKU found for selected variant');
```

Comment:
Doesn't need to happen in this pull, but it would be nice to indicate to the user that something went wrong, and report to Sentry so we know about products that have a missing sku.

Reply:
Instead of showing an error to the user, we might just hide the section. Anyway I'll rather merge this as is so I can unlock other PRs. We'll address this edge case in a separate PR